source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -326,10 +326,23 @@ void CacheFilter::getBody() {
   // posted callback.
   CacheFilterWeakPtr self = weak_from_this();
 
+  // We don't want to request more than a buffer-size at a time from the cache.
+  uint64_t fetch_size_limit = encoder_callbacks_->encoderBufferLimit();
+  // If there is no buffer size limit, we probably still want *some* constraint,
+  // because reading a gigabyte of cache entry in one go is always going to cause
+  // some kind of problem. Everyone knows 64MB should be enough for anyone.
+  if (fetch_size_limit == 0) {
+    fetch_size_limit = 64 * 1024 * 1024;
```

Comment:
https://quoteinvestigator.com/2011/09/08/640k-enough/#:~:text=%E2%80%9C640K%20ought%20to%20be%20enough%20for%20anybody.%E2%80%9D,-%E2%80%93%20Remark%20attributed%20to&text=In%20conclusion%2C%20the%20evidence%20is,was%20introduced%20in%20that%20year.

> “640K ought to be enough for anybody.”
– Remark attributed to Bill Gates (Founder and CEO of Microsoft), 1981


Anyway, actionable comment: make this a constant definition? I'm also wondering why there isn't a test that uses that to allocate FetchSizeLimit+1 and see it streamed through in 2 chunks.

-- oh I see you mock that call out. Actually it might be slightly preferable to have a setter for it rather than having a code-path (fetch_size_limit!=0) that can only be reached in tests.

I'm not going to block lgtm on this; just let me know what you think and we can move on.

Reply:
Yeah, the quote is intentionally recognizable but not accurate (to the maybe-myth), because 640K would probably be a bit small of a default chunk size when the user hasn't specified a limit, these days.

I'm not sure I understand what's being questioned though. Why is fetch_size_limit!=0 only reachable in tests? Buffer limit is a thing that I think is *usually* set to a nonzero value? And I think it *can be* set to zero-as-unlimited, so neither code-path is unreachable, but I might be wrong.

I'll do the named constant on Tuesday when I'm actually working.